src/Microsoft.TestPlatform.ObjectModel/TestCase.cs

```diff
@@ -10,13 +10,13 @@ namespace Microsoft.VisualStudio.TestPlatform.ObjectModel
     using System.Linq;
 
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities;
-    using System.Reflection;
+    using Microsoft.VisualStudio.TestPlatform.Extensions;
 
     /// <summary>
     /// Stores information about a test case.
     /// </summary>
     [DataContract]
-    public sealed class TestCase : TestObject
+    public sealed class TestCase : TestObject, ITestCase2
```

Comment:
It seems like there could be a failure in the case where a new test adapter (implementing the Managed properties), was used with an older version of the test platform.  Say, for instance, that vstest.console is run from the command line on an older install, but it is referencing a new adapter that supports the new properties. I think it would build and load properly, but the extension methods that access ITestCase2 would fail. Can we protect against that?

Reply:
The problem is that we can't access `TestProperty` from this new package. The only way I can see is not providing this interface at all and doing everything in the test adapters; in that case, I don't see a way to customize `TestId` generation. Any ideas?

Reply:
If we do everything in the test adapters, it seems to me that would work. TestId's are ideally created in the test adapter, so it seems like they could use the new code in the nuget package to generate it. Am I missing something?

Reply:
> TestId's are ideally created in the test adapter

In this case, no. I'll move the functionality into the test adapter.